### ISSUE TITLE
is it possible to flush all batched events at once?

### ISSUE THREAD

**Issue by lukehatcher** [mixpanel/mixpanel-js#426]
I know there is 

```ts
mixpanel.track('my event', {foo: 'bar'}, { 'send_immediately': true });
```

Similarly, I know I can edit the config - but that only effects following events, plus I still want batching.
```ts
mixpanel.set_config({ batch_requests: false });
// or
mixpanel.set_config({ batch_flush_interval_ms: 0 });
```

**Use Case**: I have a large `batch_flush_interval_ms` interval and want to flush everything before navigation to a new external URL, but I don't want to give up my batching and use `send_immediately` every time I call `mixpanel.track`

 I have tried editing the `flushAfter` property of the events queued inside localStorage with no luck.

**Comment by tdumitrescu** [comment]
I don't think we currently have a publicly-exposed method to flush everything. The SDK does already attempt to flush every time the page/tab loses visibility: https://github.com/mixpanel/mixpanel-js/blob/34b4396de534e4f5cf4b5cac80afd14a8322ce1f/src/mixpanel-core.js#L308-L334 but this can never be totally reliable.

If you just want something that works quickly, calling `mixpanel.start_batch_senders()` will cause them to flush immediately even if it's already running (but this is an implementation detail).

**Comment by nicolasmafraintelipost** [comment]
Some update?

**Comment by JRebella** [comment]
why isnt this a thing yet?

Its an incredibly common use case
- Need to redirect to external site and unload app? Call something on mixpanel that flushes all batched events and returns a promise (or accepts a callback) to know when its ready and safe to redirect without losing events
